src/data-workspace/use-data-value-mutation/optimistically-set-data-value-comment.js

```diff
@@ -0,0 +1,54 @@
+import addDataValue from './add-data-value.js'
+import getDataValueIndex from './get-data-value-index.js'
+import updateDataValue from './update-data-value.js'
+
+export default async function optimisticallySetDataValueComment({
+    queryClient,
+    newDataValue,
+    dataValueSetQueryKey,
+    attributeCombo,
+    attributeOptions,
+}) {
+    // Cancel any outgoing refetches (so they don't overwrite our optimistic update)
+    await queryClient.cancelQueries(dataValueSetQueryKey)
+
+    // Snapshot the previous value
+    const previousDataValueSet = queryClient.getQueryData(dataValueSetQueryKey)
+
+    // Optimistically update to the new value
+    queryClient.setQueryData(dataValueSetQueryKey, () => {
+        // dataValueSet.dataValues can be undefined
+        const previousDataValues = previousDataValueSet.dataValues || []
+        const matchIndex = getDataValueIndex(previousDataValues, newDataValue)
+        const isNewDataValue = matchIndex === -1
+
+        // If the field was previously empty the dataValue won't exist yet
+        if (isNewDataValue) {
+            const formattedNewDataValue = {
+                ...newDataValue,
+                attributeCombo,
+                attributeOptions,
+                categoryOptionCombo: newDataValue.co,
+                dataElement: newDataValue.de,
+                orgUnit: newDataValue.ou,
+                period: newDataValue.pe,
+            }
+
+            return addDataValue(previousDataValueSet, formattedNewDataValue)
+        } else {
+            const { comment } = newDataValue
+            const formattedNewDataValue = {
```

Comment:
I don't really understand why we can't consolidate all these `optimisticallyUpdate-functions`. Just transform the "new-value" before you send it to a shared `optimisticalyUpdateDataValue`-function.

Reply:
I mean, as far as I can tell the only difference between the functions was the actual updated value. I might have missed some other differences, but as far as I can see, in the end these functions are actually the same mutation - the value is just different.

Reply:
I will come back to this tomorrow, currently only have access to phone.